elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/impl/ElasticsearchIndexManager.java

```diff
@@ -499,6 +523,16 @@ public String getActualIndexName() {
 		return actualIndexName;
 	}
 
+	@Override
+	public Analyzer createAnalyzer(String name) {
+		return new ElasticsearchAnalyzer( name );
```

Comment:
I wished we'd avoid creating such "fake" implementations. The stuff gets tougher and tougher to grasp.

Did you consider to establish an abstraction, say `AnalyzerReference`. For ES, the implementation would just refer to the name, whereas for embedded Lucene there'd be another implementation referencing the actual Lucene `Analyzer` instance.

I'm not sure whether it's doable without breaking SPIs, but if it is, I'd prefer that. Otherwise it should go to the 6.0 to-do list.